backend/api.py

```
@@ -0,0 +1,62 @@
+    X_train, y_train = cbmodel.process_data(df, target_column_name=params.target_column_name)
+    cbmodel.train(X_train,
+                  y_train,
+                  iterations=params.iterations,
+                  depth=params.depth)
+
+    cbmodel.save(params.weights_name)
+
+    return params.weights_name
+
+@app.post("/predict/")
+async def predict(params: ModelPredictParams = Depends(ModelPredictParams), file: UploadFile = File(...)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)